dev.py

```diff
@@ -48,34 +49,80 @@ def cmd_all_npm_build(args):
         run_verbose(["npm", "run", "build"], cwd=str(frontend_dir))
 
 
-def cmd_all_install_python_deps(args):
-    """"Install all dependencies needed to run e2e tests for all examples and templates"""
+def cmd_e2e_build_images(args):
+    """Build docker images for each component e2e tests"""
     for project_dir in EXAMPLE_DIRECTORIES + TEMPLATE_DIRECTORIES:
-        run_verbose(["pip", "install", "-e", ".[devel]"], cwd=str(project_dir))
+        e2e_dir = next(project_dir.glob("**/e2e/"), None)
+        if e2e_dir:
+            # Define the image tag for the docker image
+            image_tag = (
+                f"component-template:py-{args.python_version}-st-{args.streamlit_version}-component-{project_dir.parts[-1]}"
+            )
+            # Build the docker image with specified build arguments
+            run_verbose(
+                [
+                    "docker",
+                    "build",
+                    ".",
+                    f"--build-arg=STREAMLIT_VERSION={args.streamlit_version}",
+                    f"--build-arg=PYTHON_VERSION={args.python_version}",
+                    f"--tag={image_tag}",
+                    "--progress=plain",
+                ],
+                env={**os.environ, "DOCKER_BUILDKIT": "1"},
+            )
 
 
-def cmd_all_install_wheel_packages(args):
-    """"Install wheel packages of all examples and templates for e2e tests"""
+def cmd_e2e_run(args):
+    """"Run e2e tests for all examples and templates in separate docker images"""
     for project_dir in EXAMPLE_DIRECTORIES + TEMPLATE_DIRECTORIES:
-        wheel_files = list(project_dir.glob("dist/*.whl"))
-        if wheel_files:
-            wheel_file = wheel_files[0]
-            run_verbose(["pip", "install", str(wheel_file)], cwd=str(project_dir))
-        else:
-            print(f"No wheel files found in {project_dir}")
-
-
-def cmd_install_browsers(args):
-    """"Install multiple browsers to run e2e for all examples and templates"""
-    run_verbose(["playwright", "install", "webkit", "chromium", "firefox", "--with-deps"])
+        container_name = project_dir.parts[-1]
+        image_tag = (
+            f"component-template:py-{args.python_version}-st-{args.streamlit_version}-component-{container_name}"
+        )
+        e2e_dir = next(project_dir.glob("**/e2e/"), None)
+        if e2e_dir:
+            mount_catalog = e2e_dir.parts[-2]
```

Comment:
Not sure why, but also create CustomDataframe for me, even if there is no tests. I would leave it for now, as we will be testing this component in the future anyway, but it will be possible to check that we are not creating an unnecessary container once we have all the tests in place

Reply:
It is possible that you had an empty catalog "e2e" in this component

Reply:
Actually I don't have, but probably it's waste of time to debug it, since we want to have this tests in the future